src/java/org/tensorics/core/booleans/lang/OngoingDetectionDirectionAndTensorAware.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2016 European Organisation for Nuclear Research (CERN), All Rights Reserved.
+ */
+
+package org.tensorics.core.booleans.lang;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.tensorics.core.tensor.Shape;
+import org.tensorics.core.tensor.Tensor;
+
+public class OngoingDetectionDirectionAndTensorAware<T extends Comparable<T>> {
+
+    private Class<T> clazz;
+    private Tensor<Boolean> tensor;
+
+    /**
+     * @param clazzToReturn
+     * @param tensor
+     */
+    public OngoingDetectionDirectionAndTensorAware(Class<T> clazzToReturn, Tensor<Boolean> tensor) {
+        this.tensor = tensor;
+        this.clazz = clazzToReturn;
+        Shape shape = tensor.shape();
+
+        if (shape.dimensionSet().size() > 1) {
```

Comment:
Fine for the first iteration, but actually I'm wondering ... wouldn't it be easy enough to implement this for any tensor, given any position as initial point, and then return an Iterable<Position>?
Of course if you need this in the DSL, it could still be a specific case, automatically infering the first element and unwrapping the Position.
Also in principle it could work on tensors of any value, not just on a Boolean I guess (using equals()).

Of course if you plan to implement some more advanced signal detection with tolerance etc. it would become much more difficult in the general case.